### ISSUE TITLE
rancher2_project_role_template_binding recreation loop when using user_principal_id

### ISSUE THREAD

**Issue by gillarda** [rancher/terraform-provider-rancher2#741]
## Versions

* Terraform 1.0.6
* Rancher provider 1.17.2
* Rancher server v2.4

## Description

I am trying to set permissions on projects for users originating for an OpenLDAP authentication source.
The inial creation of the `rancher2_project_role_template_binding` works fine 👍 

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.rancher-terraform-module.module.project["tfepsilon"].rancher2_project_role_template_binding.prtb_test will be created
  + resource "rancher2_project_role_template_binding" "prtb_test" {
      + annotations        = (known after apply)
      + group_principal_id = (known after apply)
      + id                 = (known after apply)
      + labels             = (known after apply)
      + name               = "prtb-test-people"
      + project_id         = "c-jrmjd:"p-qhacq"
      + role_template_id   = "project-member"
      + user_principal_id  = "openldap_user://cn=usertest,ou=org,ou=People,dc=domain,dc=fr"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

However,  upon subsequent `terraform plan` the resource is planned for replacement as the `rancher2_project_role_template_binding` references the `user_id` as well as the `user_principal_id` :
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # module.rancher-terraform-module.module.project["tfepsilon"].rancher2_project_role_template_binding.prtb_test must be replaced
-/+ resource "rancher2_project_role_template_binding" "prtb_test" {
      ~ annotations        = {
          - "auth.cattle.io/principal-display-name"                = "usertest"
          - "lifecycle.cattle.io/create.cluster-prtb-sync_c-jrmjd" = "true"
          - "lifecycle.cattle.io/create.mgmt-auth-prtb-controller" = "true"
        } -> (known after apply)
      + group_principal_id = (known after apply)
      ~ id                 = "p-qhacq:prtb-test-people" -> (known after apply)
      ~ labels             = {
          - "cattle.io/creator" = "norman"
        } -> (known after apply)
        name               = "prtb-test-people"
      - user_id            = "u-chpb4czuic" -> null # forces replacement
        # (3 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

I have seen the other issues related to the use of `user_principal_id` when creating `rancher2_project_role_template_binding` or `rancher2_cluster_role_template_binding` and @rawmind0 general recommendation of using `group_principal_id`, however this will not be possible for me short term.
Furthermore I am not getting 'hard' errors from the API, 'just' these recreation loops and was wondering about a workaround or a fix to this behaviour, without having the users able to login first.

Thank you for reading.

**Comment by rawmind0** [comment]
Issue reproduced. Submitted PR https://github.com/rancher/terraform-provider-rancher2/pull/743 to fix the issue

**Comment by gillarda** [comment]
Thank you for looking into this @rawmind0 !

**Comment by rawmind0** [comment]
PR #743  alredy merged. the fix will be available at next tfp release v1.20.0
